packages/next/src/lib/metadata/get-metadata-route.ts

```diff
@@ -11,14 +11,33 @@ import { normalizePathSep } from '../../shared/lib/page-path/normalize-path-sep'
  * Give it a unique hash suffix to avoid conflicts
  *
  * e.g.
- * /app/opengraph-image.tsx -> /opengraph-image
- * /app/(post)/opengraph-image.tsx -> /opengraph-image-[0-9a-z]{6}
+ * /opengraph-image -> /opengraph-image
+ * /(post)/opengraph-image.tsx -> /opengraph-image-[0-9a-z]{6}
+ *
+ * Sitemap is an exception, it should not have a suffix.
+ * As the generated urls are for indexer and usually one sitemap contains all the urls of the sub routes.
+ * The sitemap should be unique in each level and not have a suffix.
+ *
+ * /sitemap -> /sitemap
+ * /(post)/sitemap -> /sitemap
  */
 function getMetadataRouteSuffix(page: string) {
-  let suffix = ''
+  // Remove the last segment and get the parent pathname
+  // e.g. /parent/a/b/c -> /parent
+  // e.g. /parent/opengraph-image.tsx -> /parent
+  const parentPathname = page.slice(0, -(path.basename(page).length + 1))
```

Comment:
would it make more sense to use `path.dirname` here? 

```suggestion
  const parentPathname = path.dirname(page)
```